### ISSUE TITLE
not able to install with vs 2019 community version 

### ISSUE THREAD

**Issue by kunalvaibhav** [codecadwallader/codemaid#688]
### Environment
- Visual Studio version: [example 2015 Community]
- CodeMaid version: [example 10.2]
- Code language: [example C#]

### Description
Replace this text with a short description and code sample.

### Steps to recreate
1. Replace this
2. text with 
3. the steps
4. to recreate

### Current behavior
Explain what it's doing and why it's wrong.

### Expected behavior
Explain what it should be doing after it's fixed.

**Comment by kunalvaibhav** [comment]
Environment
Visual Studio version: Community 2019 16.2.5
CodeMaid version: 11.1
Code language: 
Description
not able to install with vs 2019 community version

Steps to recreate
download VS 2019 community 16.2.5 and install
download codemaid 11.1 and install
according to codemaid 11.1 description, it supports 2017 & 2019 but I had install VS 2017 & VS 2019 community versions and codemaid couldn't find these products lead to fail during installation of codemaid

Expected behavior
It should integrate with VS 2019 successfully

**Comment by kunalvaibhav** [comment]
[dd_VSIXInstaller_20191104225902_08fc.log](https://github.com/codecadwallader/codemaid/files/3805281/dd_VSIXInstaller_20191104225902_08fc.log)

**Comment by kunalvaibhav** [comment]
When I tried from 
VS 2019 -> Extensions -> Manage Extension -> Online -> Codemaid (search) -> download -> close VS 2019
[dd_VSIXInstaller_20191104230321_26a0.log](https://github.com/codecadwallader/codemaid/files/3805295/dd_VSIXInstaller_20191104230321_26a0.log)

**Comment by codecadwallader** [comment]
Thanks for reporting the issue.  The logs show:
> 04-Nov-19 11:03:34 PM - Install Error : System.InvalidOperationException: Cannot find setup engine instance.
I found this issue that indicates it was a known issue in Visual Studio that was previously resolved and required a re-install.  https://github.com/gitextensions/gitextensions/issues/5516

**Comment by kunalvaibhav** [comment]
I have VS 2017 and VS 2019 both installed in my win7 machine.
When I used CodeMaid_v11.1.209.vsix (latest), it was able to find VS 2017 and installed successfully, but it could not find VS2019.
and I want to install codemaid for VS 2019
![image](https://user-images.githubusercontent.com/15246719/68305752-3a52f680-00ce-11ea-87ba-8f013827675f.png)
![image](https://user-images.githubusercontent.com/15246719/68305853-653d4a80-00ce-11ea-898b-767c5925a417.png)
![image](https://user-images.githubusercontent.com/15246719/68305917-89009080-00ce-11ea-9284-e0bd690bfdff.png)

**Comment by codecadwallader** [comment]
Are you still getting the same error message as before?  Did the previous link help?

**Comment by kunalvaibhav** [comment]
yes, same issue in VS 2019, and the link shared was for VS 2017 where I do not have any issues in installation

**Comment by kunalvaibhav** [comment]
can some one tell me, when I can expect the fix for the above issue?

**Comment by codecadwallader** [comment]
The previous link suggested an uninstall/re-install of Visual Studio.  Have you tried that and if so did it help?  The "setup engine instance" is something outside of the scope of CodeMaid so I do not have a direct solution for it and would probably recommend contacting Microsoft support.

**Comment by Protiguous** [comment]
Codemaid 11.1 installs just fine in Visual Studio 16.4.

If you are still having issues, running "More->Repair" in the Visual Studio Installer may help?
**Warning: this will reset any Visual Studio configurations you've set.**

![image](https://user-images.githubusercontent.com/37120672/70539239-a3d88180-1b20-11ea-8ef6-f979a6e14be7.png)

**Comment by clacina** [comment]
The download gives me a  .vslx file.  I can't figure out how to get visual studio 2019 to recognize this.  Also, searching the repo in the extension manager does not find CodeMaid.  What am I missing?

**Comment by codecadwallader** [comment]
@clacina If you have a .vsix file you can usually double click it to start and it will attempt to install into applicable Visual Studio versions.  You do need have to have .NET Framework v4.7.2 or higher installed for CodeMaid v11.1

**Comment by michaelm7456** [comment]
I've just confirmed that CodeMaid installs correctly on my instance of Visual Studio 2019.

Microsoft Visual Studio Community 2019
Version 16.6.4

I had encountered the '**Cannot find setup engine instance**' issue and found that by installing a missing Visual Studio Workload that this resolved the issue.

Via the '**Visual Studio Installer**' app (Check the '**Visual Studio Extension Development**' Workload is installed.)

[Stack Overflow answer which resolved this for me]
https://stackoverflow.com/questions/54382367/visual-studio-extension-installation-cannot-find-setup-engine-instance/56001724#56001724

[Image]
https://i.stack.imgur.com/UMg16.jpg

Hope this helps!

**Comment by codecadwallader** [comment]
@michaelm7456 Thanks for providing that workaround.  It shouldn't be necessary for most people to install the Visual Studio Extension Development workload unless they are planning to work on extensions, but regardless I'm glad it helped in your case and for the tip in case it helps others!
